### ISSUE TITLE
POC - Refactoring XLF catalog

### ISSUE THREAD

**Issue by ibahloul-ps** [PrestaShop/PrestaShop#37072]
Change the format of the XLF file as follow : 
```
<?xml version="1.0" encoding="UTF-8"?>
<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
  <trans-unit id="b2f40690858b404ed10e62bdf422c704">
    <source>Amount</source>
    <target>Amount</target>
  </trans-unit>
  <trans-unit id="8c2857a9ad1d8f31659e35e904e20fa6">
    <source>Logo</source>
    <target>Logo</target>
  </trans-unit>
  <trans-unit id="deb10517653c255364175796ace3553f">
    <source>Product</source>
    <target>Product</target>
  </trans-unit>
  <trans-unit id="befcac0f9644a7abee43e69f49252ac4">
    <source>Tax excluded</source>
    <target>Tax excluded</target>
  </trans-unit>
  <trans-unit id="f4a0d7cb0cd45214c8ca5912c970de13">
    <source>Tax included</source>
    <target>Tax included</target>
  </trans-unit>
  <trans-unit id="689202409e48743b914713f96d93947c">
    <source>Value</source>
    <target>Value</target>
  </trans-unit>
</xliff>
```
Next steps :

1. Check that the proposal is a valid XLF format
2. If valid, have the [translation tool](https://github.com/PrestaShopCorp/TranslationTool/) generate the file in the new format
3. Update the default catalog of the Core
4. /!\ Manually revalidate translations (in all languages) in Crowndin

**Comment by Hlavtox** [comment]
@ibahloul-ps @jolelievre Sooo, does this mean that the filenames are removed? It may be useful to check the context using the file when translating. 🤔

**Comment by jolelievre** [comment]
@Hlavtox yeah the idea is to remove the filename because this is what causes most of our issues, we won't have a perfect solution anyway, and it's convenient for you because you're a developer and can easily navigate in the code to understand the context but I think for most translators this info doesn't help so it bring more harm than good

Maybe we can try an alternative like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
  <trans-unit id="b2f40690858b404ed10e62bdf422c704">
    <source>Amount</source>
    <target>Amount</target>
    <note>MyPHPClass:L167</note>
  </trans-unit>
</xliff>
```

But that's only if:
- it's doable technically
- it means the note sub node will also be modified a lot an will pollute the PRs, and mostly we need to check if such modification also triggers Crowdin to consider this a new wording, if it does then we'd better not integrate it

The idea behind this is first to have a POC in the export tool to see if it's doable, then check if this format is accepted as valid XLF and understandable by PrestaShop and finally we will test the push to crowdin in a dedicated test branch to see how it reacts.

**Comment by Hlavtox** [comment]
@jolelievre I won't mind, if it fixes the issue, but what about keeping the filenames and just ensuring the order will be consistent? :-)

**Comment by jolelievre** [comment]
It won't work if we keep the list of files, as I explained to you earlier the problem comes from this split between files, even if we order the list of files alphabetically AND inside those nodes we also sort the wordings alphabetically it won't change the fact that if a wording is added in a new file, and this file is the first one to be matched with the wording, then the wording will be changed from file node (removed from previous file node matching, added in new file node matching).

This is what causes all the useless modifications we have now, even if we don't understand the order of matching and it depends on the environment the structure of the file is the core problem and I don't see another solution but to change it.

**Comment by jolelievre** [comment]
Both the core and the TranslationTool have been updated now, all that is left is to test the Crowdin update
